core/src/main/scala/org/apache/spark/storage/BlockManagerId.scala

```diff
@@ -69,24 +72,37 @@ class BlockManagerId private (
     out.writeUTF(executorId_)
     out.writeUTF(host_)
     out.writeInt(port_)
+    out.writeBoolean(topologyInfo_.isDefined)
+    // we only write topologyInfo if we have it
+    topologyInfo.foreach(out.writeUTF(_))
   }
 
   override def readExternal(in: ObjectInput): Unit = Utils.tryOrIOException {
     executorId_ = in.readUTF()
     host_ = in.readUTF()
     port_ = in.readInt()
+    val isTopologyInfoAvailable = in.readBoolean()
+    topologyInfo_ = if (isTopologyInfoAvailable) {
```

Comment:
it might be more clear to do
```
if (isTopologyInfoAvailable) {
  topologyInfo_ = Option(in.readUTF())
}  else {
  topologyInfo_ = None
}
```

or

```
topologyInfo_ = if (isTopologyInfoAvailable) Option(in.readUTF()) else None
```